src/main/scala/uk/gov/nationalarchives/omega/api/messages/LocalMessage.scala

```diff
@@ -0,0 +1,227 @@
+  *   The message payload, usually in JSON format.
+  * @param omgMessageTypeId
+  *   The Identifier of the Omega Message Type. In AsyncAPI Schema this is the messageId property of a message defined
+  *   in the schema.
+  * @param jmsMessageId
+  *   A unique identifier for the message. We suggest using a short application instance specific prefix followed by a
+  *   unique identifier. In the extreme case a UUID may be used. Depending on your JMS provider, this may be set on your
+  *   behalf.
+  * @param omgApplicationId
+  *   Identifier of the application that is publishing the message to Omega.
+  * @param jmsTimestamp
+  *   The time the message was sent. The value of the timestamp is the amount of time, measured in milliseconds, that
+  *   has elapsed since midnight, January 1, 1970, UTC. Depending on your JMS provider, this may be set on your behalf.
+  * @param omgMessageFormat
+  *   The Internet Media-Type indicating the format of the message body.
+  * @param omgToken
+  *   An Authentication Token that is valid for the OMGApplicationID and request.
+  * @param omgReplyAddress
+  *   The address that the reply to this message should be sent to. Typically this is the name of a Queue setup for a
+  *   specific application and must be prefixed with the OMGApplicationID of the application that is making the request.
+  */
+@SerialVersionUID(1L)
+final case class LocalMessage(
+  persistentMessageId: Version1UUID,
+  messageText: String,
+  omgMessageTypeId: Option[String],
+  jmsMessageId: Option[String],
+  omgApplicationId: Option[String],
+  jmsTimestamp: Option[Long],
+  omgMessageFormat: Option[String],
+  omgToken: Option[String],
+  omgReplyAddress: Option[String]
+) extends Serializable {
+
+  def validateOmgApplicationId: ValidationResult[String] =
+    omgApplicationId match {
+      case Some(applicationId) if patternForApplicationId.matches(applicationId) => applicationId.validNec
+      case Some(_)                                                               => InvalidApplicationID.invalidNec
+      case None                                                                  => MissingApplicationID.invalidNec
+    }
+
+  def validateOmgToken: ValidationResult[String] =
+    omgToken match {
+      case Some(token) if token.trim.nonEmpty => token.validNec
+      case Some(_)                            => InvalidAuthToken.invalidNec
+      case None                               => MissingAuthToken.invalidNec
+    }
+
+  def validateOtherHeaders(
+    validatedOmgToken: ValidationResult[String],
+    validatedOmgApplicationId: ValidationResult[String]
+  ): ValidationResult[ValidatedLocalMessage] =
+    (
+      validateJmsMessageId(jmsMessageId),
+      validateOmgMessageTypeId,
+      validatedOmgApplicationId,
+      validateJmsTimestamp(jmsTimestamp),
+      validateOmgMessageFormat(omgMessageFormat),
+      validatedOmgToken,
+      validateOmgResponseAddress(omgReplyAddress, omgApplicationId)
```

Comment:
should this method name be renamed to `validateOmgReplyAddress` to match the input parameter `replyAddress`

Reply:
Yes it should. Amended. I had a discussion with @adamretter about this on another ticket and we agreed that "reply" is semantically better for async messaging than "response" as it suggests a more considered answer (which is what we are aiming for) rather than just an rapid acknowledgement. It is also fits with the terminology in [the design pattern we are using](https://www.enterpriseintegrationpatterns.com/patterns/messaging/RequestReply.html).

Reply:
OK